src/System.Linq/tests/AllTests.cs

```diff
@@ -14,90 +13,58 @@ public class AllTests : EnumerableTests
         public void SameResultsRepeatCallsIntQuery()
         {
             var q = from x in new[] { 9999, 0, 888, -1, 66, -777, 1, 2, -12345 }
-                    where x > Int32.MinValue
+                    where x > int.MinValue
                     select x;
 
             Func<int, bool> predicate = IsEven;
-
             Assert.Equal(q.All(predicate), q.All(predicate));
         }
 
         [Fact]
         public void SameResultsRepeatCallsStringQuery()
         {
-            var q = from x in new[] { "!@#$%^", "C", "AAA", "", "Calling Twice", "SoS", String.Empty }
+            var q = from x in new[] { "!@#$%^", "C", "AAA", "", "Calling Twice", "SoS", string.Empty }
                     select x;
 
-            Func<string, bool> predicate = String.IsNullOrEmpty;
-
+            Func<string, bool> predicate = string.IsNullOrEmpty;
             Assert.Equal(q.All(predicate), q.All(predicate));
         }
 
-        [Fact]
-        public void SourceIsEmpty()
-        {
-            int[] source = { };
-
-            Assert.True(source.All(IsEven));
-        }
-
-        [Fact]
-        public void OneElementPredicateFalse()
-        {
-            int[] source = { 3 };
-
-            Assert.False(source.All(IsEven));
-        }
-
-        [Fact]
-        public void OneElementPredicateTrue()
-        {
-            int[] source = { 4 };
-
-            Assert.True(source.All(IsEven));
-        }
-
-        [Fact]
-        public void PredicateTrueOnSomeButNotAll()
+        public static IEnumerable<object[]> All_TestData()
         {
-            int[] source = { 4, 8, 3, 5, 10, 20, 12 };
-
-            Assert.False(source.All(IsEven));
+            Func<int, bool> isEvenFunc = IsEven;
+            yield return new object[] { new int[0], isEvenFunc, true };
+            yield return new object[] { new int[] { 3 }, isEvenFunc, false };
+            yield return new object[] { new int[] { 4 }, isEvenFunc, true };
+            yield return new object[] { new int[] { 3 }, isEvenFunc, false };
+            yield return new object[] { new int[] { 4, 8, 3, 5, 10, 20, 12 }, isEvenFunc, false };
+            yield return new object[] { new int[] { 4, 2, 10, 12, 8, 6, 3 }, isEvenFunc, false };
+            yield return new object[] { new int[] { 4, 2, 10, 12, 8, 6, 14 }, isEvenFunc, true };
+
+            int[] range = Enumerable.Range(1, 10).ToArray();
```

Comment:
nit: It looks like you don't necessarily need to use `ToArray` since the test case takes an `IEnumerable`. (Doesn't really matter either way)